csrc/ops/alias.cpp

```diff
@@ -317,16 +316,26 @@ TensorView* permute(TensorView* x, const std::vector<int64_t>& new2old) {
   auto normalized_new2old =
       ir_utils::normalizeNew2Old(new2old, inp_domain.size());
 
-  for (const auto i : c10::irange(out_domain.size())) {
-    auto in_id = inp_domain[normalized_new2old[i]];
-    out_domain[i] = in_id->cloneWithoutRFactor();
+  std::vector<IterDomain*> out_root;
+  out_root.reserve(inp_domain.size());
+  for (const auto id : inp_domain) {
+    out_root.emplace_back(id->cloneWithoutRFactor());
+  }
+
+  std::vector<IterDomain*> out_rfactor;
+  out_rfactor.reserve(inp_domain.size());
+  for (const auto i : normalized_new2old) {
+    out_rfactor.emplace_back(out_root.at(i));
```

Comment:
One thing that's different from other rfactor-based ops is that the rfactor IDs are not marked as rfactor. I'm not sure if it might cause any issue. There's no expression here between the root and rfactor domains, so maybe it doesn't matter if it's marked as rfactor or not. Still, the change makes me a little nervous.

Reply:
Hmm, I feel it makes more sense to not mark it as rfactor rather than marking it as rfactor. I will feel more nervous if it is marked as rfactor. I don't have more concrete reason why I have this feeling, maybe this comes from the fact the for a reduction rfactor, only the IDs that is transformed get this mark.

Reply:
Yeah, that also makes sense. Feeling better now.